scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java

```diff
@@ -817,5 +832,15 @@ private void determineCipherSuitesFromConfig() {
 
 			config.supportedCipherSuites = ciphers.toArray(new CipherSuite[0]);
 		}
+	
+
+		public void setTrustedRPKs(Set<RawPublicKeyIdentity> rpks) {
+		    if (rpks == null || rpks.isEmpty()) {
+		        throw new IllegalStateException("Must provide a non-null, non-empty"
+		                + " set of trusted raw public keys"); 
+		    }
+		    config.trustedRPKs = new HashSet<>();
+		    config.trustedRPKs.addAll(rpks);
```

Comment:
If this is intended to be "unmodifiable", please use `Collections.unmodifiableSet()`. Then all consumer may just use it without cloning the outer `Set`.

Reply:
FMPOV: a interface would be really a better solution. Then `Collections.unmodifiableSet()` would be obsolete.